dcap/types/src/certification_data.rs

```diff
@@ -16,14 +16,45 @@ type Result<T> = ::core::result::Result<T, Quote3Error>;
 /// Table 9 of
 /// <https://download.01.org/intel-sgx/latest/dcap-latest/linux/docs/Intel_SGX_ECDSA_QuoteLibReference_DCAP_API.pdf>.
 #[derive(Debug, Clone, Hash, PartialEq, Eq)]
-pub struct CertificationData<'a> {
-    // The `Certification Data` field as described in the QuoteLibReference.
-    // The length of this *will* equal the `size` field as described in the
-    // QuoteLibReference
+pub enum CertificationData<'a> {
+    /// Contains:
+    /// - Platform provisioning ID(PPID)
+    /// - CPU security version number (CPUSVN)
+    /// - Provisioning certification enclave security version number (PCESVN)
+    /// - Provisioning certification enclave ID (PCEID)
+    Ppid(&'a [u8]),
+
+    /// Contains the following encrypted with RSA 2048:
+    /// - Platform provisioning ID(PPID)
+    /// - CPU security version number (CPUSVN)
+    /// - Provisioning certification enclave security version number (PCESVN)
+    /// - Provisioning certification enclave ID (PCEID)
+    PpidEncryptedRsa2048(&'a [u8]),
+
+    /// Contains the following encrypted with RSA 3072:
+    /// - Platform provisioning ID(PPID)
+    /// - CPU security version number (CPUSVN)
+    /// - Provisioning certification enclave security version number (PCESVN)
+    /// - Provisioning certification enclave ID (PCEID)
+    PpidEncryptedRsa3072(&'a [u8]),
+
+    /// Contains the provisioning certification key leaf certificate
+    Pck(&'a [u8]),
+
+    /// Contains the provisioning certification key chain
+    PckCertChain(PckCertChain<'a>),
```

Comment:
The use of `Cert` here deviates from the outer type using `Certification`.  I left it short to see what peoples thoughts were on shortening the use of `Certification`in types.
I'm up for suggestions:
- Spell this one out
- Rename the outer (thinking this would be best done in a follow on PR).

Reply:
+1 to spelling it out in order to maintain consistency.

Reply:
Spelled it out, it's actually `Certificate` in this instance so spelling out might be better to disambiguate